SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,233 @@
+//==------- ctor_load_core.cpp  - DPC++ ESIMD on-device test ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for simd load constructor.
+// The test uses reference data and different alignment flags. Invokes simd
+// constructors in different contexts with provided reference data and alignment
+// flag.
+// It is expected for destination simd instance to store a bitwise same data as
+// the reference one.
+
```

Comment:
Nits (can be added later)
- accessor-based load constructor not tested? If tested elsewhere, this should be called smth like usm_ctor*...
- most popular alignment for loading as a block will be overaligned<16>{} - oword. This is minimum requirement for block load. Not tested.

Reply:
> accessor-based load constructor not tested


We are going to add tests on accessor-based load constructor in separate PR

> most popular alignment for loading as a block will be overaligned<16>{} - oword. This is minimum requirement for block load. Not tested.


For overaligned we are using `std::max_align_t` that means we will use synonymous with the largest scalar type, which is `long double` on most platforms.
In the test we defined it [here](https://github.com/intel/llvm-test-suite/pull/769/files#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7R112-R119)
Launch tests with this alignment type [here](https://github.com/intel/llvm-test-suite/pull/769/files#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7R226)

Reply:
so please extend the testcases with oword alignment.

Reply:
Alignment will be changed in one of next PRs